mephisto/core/operator.py

```diff
@@ -290,6 +290,11 @@ def _track_and_kill_runs(self):
             runs_to_check = list(self._task_runs_tracked.values())
             for tracked_run in runs_to_check:
                 task_run = tracked_run.task_run
+                if not tracked_run.task_launcher.assignment_thread_done:
+                    continue
+                else:
+                    task_run.assignment_generators_done = True
```

Comment:
It's somewhat strange that the operator is directly checking and setting attributes for the task run. The semantics here would be better in calling `task_run.update_completion_progress(task_launcher)` or something similar, and then let the `get_is_completed()` call still be responsible for the implicit continue in this loop. The `update_completion_progress` method should ideally only rely on `task_launcher`'s functions, rather than direct accesses to its attributes.